pkg/schedule/cron_schedule.go

```diff
@@ -0,0 +1,121 @@
+/*
+http://www.apache.org/licenses/LICENSE-2.0.txt
+
+Copyright 2016 Intel Corporation
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package schedule
+
+import (
+	"errors"
+	"time"
+
+	"github.com/robfig/cron"
+)
+
+var ErrMissingCronEntry = errors.New("Cron entry is missing")
+
+// CronSchedule is a schedule that waits as long as specified in cron entry
+type CronSchedule struct {
+	entry    string
+	state    ScheduleState
+	schedule *cron.Cron
+}
+
+// NewCronSchedule creates and starts new cron schedule and returns an instance of CronSchedule
+func NewCronSchedule(entry string) *CronSchedule {
+	schedule := cron.New()
+	schedule.Start()
```

Comment:
This cron scheduler introduces a difference in implementation from the simple and windowed schedule types since simple is just an interval from the time the task starts to the time the task stops, and windowed is running at a certain interval for a period of time. 

The implementation of the cron package being used for the cron schedule has actual methods for starting and stopping that are not timed based. The schedule should not start when the schedule is created, but when the task is started, and the schedule should stop when the task is stopped as the schedule is only valid when the task is running. In this instance, issuing `schedule.Start()` when the schedule is created is firing off a go routine with no ability to stop it without killing snapd. This may require adding additional methods for simple and windowed schedules `Start()` and `Stop()` respectively that for those schedule do nothing, but for cron schedule would actually start and stop the schedule. This would allow `task.Start()` and `task.Stop()` to call start and stop on any schedule without having to check if every task is a cron schedule or not.